Source/WebCore/Modules/WebGPU/Implementation/WebGPUXRBindingImpl.cpp

```diff
@@ -60,15 +60,15 @@ RefPtr<XRSubImage> XRBindingImpl::getSubImage(XRProjectionLayer&, WebCore::WebXR
     return nullptr;
 }
 
-RefPtr<XRSubImage> XRBindingImpl::getViewSubImage(XRProjectionLayer& projectionLayer, XREye eye)
+RefPtr<XRSubImage> XRBindingImpl::getViewSubImage(XRProjectionLayer& projectionLayer)
 {
-    UNUSED_PARAM(projectionLayer);
-    return XRSubImageImpl::create(adoptWebGPU(wgpuBindingGetViewSubImage(m_backing.get(), m_convertToBackingContext->convertToBacking(eye))), m_convertToBackingContext);
+    auto& projectionLayerImpl = static_cast<XRProjectionLayerImpl&>(projectionLayer);
+    return XRSubImageImpl::create(adoptWebGPU(wgpuBindingGetViewSubImage(m_backing.get(), projectionLayerImpl.backing())), m_convertToBackingContext);
 }
 
 TextureFormat XRBindingImpl::getPreferredColorFormat()
 {
-    return TextureFormat::Bgra8unorm;
+    return TextureFormat::Bgra8unormSRGB;
```

Comment:
This might be an issue. In WebGL-based WebXR we did this and the WebGL spec assumes that it's Bgra8unorm by default so the rendering was coming out lighter than expected.

Reply:
In theory this would be a website bug as the website should use the color format from getPreferredColorFormat() according to https://github.com/immersive-web/WebXR-WebGPU-Binding/blob/main/explainer.md#webgpu-binding

Does the WebXR WebGL variant have a similar preferred texture format function? I couldn't find it in https://www.w3.org/TR/webxr/ but I am not so familiar